src/Workspaces/Core/Portable/Shared/Extensions/ContextQuery/AbstractSyntaxContext.cs

```diff
@@ -80,21 +82,27 @@ private ISet<INamedTypeSymbol> ComputeOuterTypes(CancellationToken cancellationT
                 var containingType = enclosingSymbol.GetContainingTypeOrThis();
                 if (containingType != null)
                 {
-                    return containingType.GetContainingTypes().ToSet();
+                    return new Tuple<ISet<ITypeSymbol>, ITypeSymbol>(containingType.GetContainingTypesAndThis().ToSet(), containingType);
```

Comment:
This smells fishy. Why do we need to store `containingType` *and* the results of `GetContainingTypesAndThis(...)`? Isn't `containingType` already included in the set?